x/feegrant/keeper/keeper.go

```diff
@@ -61,6 +73,26 @@ func NewKeeper(env appmodule.Environment, cdc codec.BinaryCodec, addrCdc address
 
 // GrantAllowance creates a new grant
 func (k Keeper) GrantAllowance(ctx context.Context, granter, grantee sdk.AccAddress, feeAllowance feegrant.FeeAllowanceI) error {
+	migrated, err := k.isXAccount(ctx, granter)
```

Comment:
This is a big downsite of migrating to xaccounts. Now we have to check for an existing xaccount all the time.

Reply:
I put the check on top as this was quicker to build but it would probably be more efficient to check state in the module first. 
fyi: I also had a "migrated accounts" index in the beginning but accounts can be created in xaccounts without migration, so this did not work